### PR TITLE
Get/create instance

### PR DESCRIPTION
**Pull request by NilayYadav**
my current code basically getting and creating containers is this implementation of container management right? I've just iterated on the `endpoint` code

### CODE REVIEW

**Comment by luke-lombardi** [comment]
Not sure what this is for. Re-open if you have a particular issue you're working on.

**Comment by NilayYadav** [comment]
Hi @luke-lombardi I'm working on implementing a new feature for Beam called "beam shell." The idea is to allow customers to SSH into containers. 

Currently the PR manage containers, getting and creating containers. Could you please review it to see if the implementation for container management is on the right track?

File: pkg/abstractions/shell/shell.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+	// Monitor and then clean up the instance once it's done
+	go instance.Monitor()
+	go func(i *shellInstance) {
+		<-i.Ctx.Done()
+		ss.shellInstances.Delete(stubId)
+	}(instance)
+
+	return instance, nil
+}
```
Comment (jsun-m) [inline pkg/abstractions/shell/shell.go:130]: You should take a look at how to use this to send commands to the container
pkg/common/runc_client.go

**Review by jsun-m — COMMENTED** [review]
an initial review.

Also please add a description and a change the name of the PR to be the feature that you are working on.

**Pull request closed without merging**